airflow/hooks/mysql_hook.py

```diff
@@ -114,8 +111,46 @@ def get_conn(self):
             conn_config['unix_socket'] = conn.extra_dejson['unix_socket']
         if local_infile:
             conn_config["local_infile"] = 1
-        conn = MySQLdb.connect(**conn_config)
-        return conn
+        return conn_config
+
+    def _get_conn_config_mysql_connector_python(self, conn):
+        conn_config = {
+            'user': conn.login,
+            'password': conn.password or '',
+            'host': conn.host or 'localhost',
+            'database': self.schema or conn.schema or ''
+        }
+
+        if not conn.port:
+            conn_config['port'] = 3306
+        else:
+            conn_config['port'] = int(conn.port)
+
+        if conn.extra_dejson.get('allow_local_infile', False):
+            conn_config["allow_local_infile"] = True
+
+        return conn_config
+
+    def get_conn(self):
+        """
+        Establishes a connection to a mysql database
+        by extracting the connection configuration from the Airflow connection.
+
+        .. note:: By default it connects to the database via the mysqlclient library.
+            But you can also choose the mysql-connector-python library which lets you connect through ssl
+            without any further ssl parameters required.
+
+        :return: a mysql connection object
+        """
+        conn = self.connection or self.get_connection(self.mysql_conn_id)  # pylint: disable=no-member
```

Comment:
Out of curiosity, what does rise `no-member` error here?

Reply:
Because `mysql_conn_id` is only accessible because of `conn_name_attr = 'mysql_conn_id'`